src/reducers/breakpoints.js

```diff
@@ -159,6 +105,108 @@ function update(state = State(), action: Action) {
   return state;
 }
 
+function addBreakpoint(state, action) {
+  const id = makeLocationId(action.breakpoint.location);
+
+  if (action.status === "start") {
+    let bp = state.breakpoints.get(id) || action.breakpoint;
+
+    return state.setIn(["breakpoints", id], updateObj(bp, {
+      disabled: false,
+      loading: true,
+      // We want to do an OR here, but we can't because we need
+      // empty strings to be truthy, i.e. an empty string is a valid
+      // condition.
+      condition: firstString(action.condition, bp.condition)
+    }))
+    .set("breakpointsDisabled", false);
+  }
+
+  if (action.status === "done") {
+    const { id: breakpointId, text } = action.value;
+    let location = action.breakpoint.location;
+    let { actualLocation } = action.value;
+
+    // If the breakpoint moved, update the map
+    if (locationMoved(location, actualLocation)) {
+      state = state.deleteIn(["breakpoints", id]);
+
+      const movedId = makeLocationId(actualLocation);
+      const currentBp = (state.breakpoints.get(movedId) ||
+                         fromJS(action.breakpoint));
+      const newBp = updateObj(currentBp, { location: actualLocation });
+      state = state.setIn(["breakpoints", movedId], newBp);
+      location = actualLocation;
+    }
+
+    const locationId = makeLocationId(location);
+    const bp = state.breakpoints.get(locationId);
+    return state.setIn(["breakpoints", locationId], updateObj(bp, {
+      id: breakpointId,
+      disabled: false,
+      loading: false,
+      text: text
+    }));
+  }
+
+  if (action.status === "error") {
+    // Remove the optimistic update
+    return state.deleteIn(["breakpoints", id]);
+  }
+}
+
+function removeBreakpoint(state, action) {
+  if (action.status === "done") {
+    const id = makeLocationId(action.breakpoint.location);
+
+    if (action.disabled) {
+      const bp = state.breakpoints.get(id);
+      const updatedState = state.setIn(["breakpoints", id], updateObj(bp, {
+        loading: false, disabled: true
+      }));
+
+      return updatedState.set(
+        "breakpointsDisabled", allBreakpointsDisabled(updatedState)
+      );
+    }
+
+    const updatedState = state.deleteIn(["breakpoints", id]);
+
+    return updatedState.set(
+      "breakpointsDisabled", allBreakpointsDisabled(updatedState)
+    );
+  }
+
+  return state;
+}
+
+function makePendingBreakpoint(bp: any) {
+  const {
+    location: { sourceUrl, line },
+    condition,
+    disabled
+  } = bp;
+
+  const location = { sourceUrl, line };
+  return { condition, disabled, location };
+}
+
+function filterByNotLoading(bp: any): boolean {
```

Comment:
I wrote this function to fix this error:
<img width="608" alt="screen shot 2017-03-10 at 11 45 20 am" src="https://cloud.githubusercontent.com/assets/4562118/23784284/0349b230-059c-11e7-860a-803cbe2ee4e0.png">